impala/hiveserver2.py

```diff
@@ -322,13 +322,26 @@ def fetchmany(self, size=None):
                 break
         return local_buffer
 
-    def fetchall(self):
+    def fetchall(self, columnar=False):
```

Comment:
I feel like `columnar` (or something a bit more descriptive) should probably be a field of `HiveServer2Cursor` that gets set on instantiation, no?  It seems a bit weird to me that someone could have a cursor and read columns or rows on different calls.  What do you think?

Another alternative might be to have a totally separate "internal-ish" method that does the columnar fetch, meant only for consumers that want to integrate more deeply (e.g., pandas).